homeassistant/components/openuv/__init__.py

```diff
@@ -125,21 +194,35 @@
             notification_id=NOTIFICATION_ID)
         return False
 
-    for component, schema in [
-            ('binary_sensor', conf[CONF_BINARY_SENSORS]),
-            ('sensor', conf[CONF_SENSORS]),
-    ]:
-        hass.async_create_task(
-            discovery.async_load_platform(
-                hass, component, DOMAIN, schema, config))
+    for component in 'binary_sensor', 'sensor':
+        hass.async_create_task(hass.config_entries.async_forward_entry_setup(
+            config_entry, component))
 
     async def refresh_sensors(event_time):
         """Refresh OpenUV data."""
         _LOGGER.debug('Refreshing OpenUV data')
         await openuv.async_update()
         async_dispatcher_send(hass, TOPIC_UPDATE)
 
-    async_track_time_interval(hass, refresh_sensors, conf[CONF_SCAN_INTERVAL])
+    data[DATA_OPENUV_LISTENER][config_entry.title] = async_track_time_interval(
+        hass, refresh_sensors,
+        conf.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL))
+
+    return True
+
+
+async def async_unload_entry(hass, config_entry):
+    """Unload an OpenUV config entry."""
+    data = hass.data[DOMAIN]
+
+    data[DATA_OPENUV_CLIENT].pop(config_entry.title)
+
+    remove_listener = data[DATA_OPENUV_LISTENER].pop(config_entry.title)
+    remove_listener()
+
+    for remove_dispatcher in data[DATA_OPENUV_DISPATCH][config_entry.title]:
```

Comment:
Don't do this. Instead, use `self.async_on_remove()` inside the `async_added_to_hass` callback.